sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/IndexedFile.java

```diff
@@ -52,15 +54,19 @@
    * This is not canonical path. Symbolic links are not resolved. For example if /project/src links
    * to /tmp/src and basedir is /project, then this method returns /project/src/index.php. Use
    * {@code file().getCanonicalPath()} to resolve symbolic link.
+   * @deprecated since 6.6 use {@link #inputStream()}, {@link #filename()} or {@link #uri()}
    */
+  @Deprecated
   @Override
   String absolutePath();
 
   /**
    * The underlying absolute {@link java.io.File}. It should not be used to read the file in the filesystem.
    * @see #contents()
    * @see #inputStream()
+   * @deprecated since 6.6 use {@link #inputStream()}, {@link #filename()} or {@link #uri()}
    */
+  @Deprecated
   @Override
   File file();
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)